### PR TITLE
Replace ZIPFoundation with native MiniZip extractor

### PR DESCRIPTION
**Pull request by glenn-jocher**
## Summary

Removes the **ZIPFoundation** third-party dependency and replaces it with **`MiniZip`** — a self-contained ZIP reader built only on Apple's **Foundation** and **Compression** frameworks. This also eliminates the SwiftPM git-clone checkout of ZIPFoundation (whose upstream `development` branch was showing up as a stray "remote branch" in IDEs).

The key enabler: ZIP method-8 entries store a **raw DEFLATE** stream, which Apple's `COMPRESSION_ZLIB` decodes directly — so no external code is required to unpack the model archives the SDK downloads.

### Why not Apple Archive (`.aar`)?
Apple Archive is Apple-proprietary with no Linux/cross-platform tooling, so it would break the Ultralytics export pipeline and any non-macOS developer. Keeping standard `.zip` and swapping only the *reader* avoids that entirely.

## Changes
- **`Sources/YOLO/MiniZip.swift`** (new): parses the central directory; extracts `STORED` (method 0) and `DEFLATE` (method 8) entries; verifies each entry's **CRC-32**; guards against **Zip Slip** path traversal; rejects encryption / ZIP64 / unknown methods rather than mis-extracting.
- **`Sources/YOLO/YOLOModelDownloader.swift`**: `unzipSkippingMacOSX` now calls `MiniZip`, preserving the existing `__MACOSX/` and `._` skip behavior. Drops `import ZIPFoundation`.
- **`Package.swift`**: removes the ZIPFoundation dependency and its target linkage.
- **`Tests/YOLOTests/MiniZipTests.swift`** (new): self-contained tests using a base64 archive fixture generated by Python's `zipfile` (an *independent* PKZIP implementation) — covers DEFLATE/STORED round-trip, nested directories, macOS-metadata skipping, and malformed/truncated input.

## Testing
- ✅ `MiniZipTests` (4 tests) pass on the iOS Simulator (arm64, iOS 26.5).
- ✅ Parity verified against Python-generated archives: byte-exact round-trip of DEFLATE + STORED entries, correct nested-directory creation, `__MACOSX`/`._` skipping.
- ✅ Malformed inputs (random bytes, truncated archive, empty file) throw cleanly instead of crashing.

> Note: on-device XCTest isn't available for this SwiftPM package (test targets have no host app — "tool-hosted testing is unavailable on device destinations"). The simulator runs the same arm64 iOS build and Compression framework. The change is pure Foundation/Compression logic with no ANE/GPU/device-specific path.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
🧩 This PR removes the app’s ZIPFoundation dependency and replaces it with a built-in pure Swift ZIP extractor, making the iOS app and YOLO package fully zero-dependency.

### 📊 Key Changes
- 🔄 Replaced third-party `ZIPFoundation` with a new internal `MiniZip.swift` extractor built on Apple’s `Foundation` and `Compression` frameworks.
- 🧹 Removed `ZIPFoundation` from `Package.swift`, Xcode project settings, and app/package imports.
- 📦 Updated both the Swift package and iOS app model download flows to use `MiniZip.extract(...)` for unpacking downloaded model archives.
- 🛡️ Added safety checks in the new extractor for:
  - invalid or corrupt ZIP files
  - unsupported ZIP features like encryption and ZIP64
  - path traversal attacks (“Zip Slip”)
  - suspiciously large decompression claims that could indicate ZIP bomb attacks
  - CRC validation to verify extracted file integrity
- ✅ Added a dedicated `MiniZipTests.swift` test suite covering normal extraction, skipped macOS metadata files, corrupt/truncated archives, empty files, comment edge cases, and decompression bomb rejection.
- 📝 Updated README files to highlight the new “Zero Dependencies” design across the repo.

### 🎯 Purpose & Impact
- 🚀 Simplifies setup and builds by removing an external package that had to be downloaded, resolved, and maintained.
- 🍎 Makes the project more Apple-native by relying only on first-party frameworks.
- 🔒 Improves security and reliability when downloading and extracting model files from remote URLs.
- 📱 Reduces dependency risk for app developers, which can help with maintenance, auditing, and licensing concerns.
- 🧠 Keeps model download functionality intact while making the codebase more self-contained and easier to ship in environments where minimizing dependencies matters.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @glenn-jocher, thank you for submitting a `ultralytics/yolo-ios-app` 🚀 PR! This is an automated message to help streamline review, and an engineer will assist shortly. Please review the checklist below for a smooth integration ✨

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/yolo-ios-app/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/yolo-ios-app` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Good simplification overall, and the safety checks/tests are a strong start. I found three issues worth addressing before merge: EOCD detection can reject valid ZIPs with comments, CRC verification is skipped for legitimate `0` CRC values, and DEFLATE extraction currently trusts uncompressed sizes enough to allow memory-exhaustion on malicious archives.

💬 Posted 3 inline comments

File: Sources/YOLO/MiniZip.swift [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+      throw MiniZipError.corruptArchive
+    }
+    return output
+  }
+
+  /// Inflates a raw DEFLATE stream to `expectedSize` bytes using the Compression framework.
+  private static func inflate(_ input: Data, expectedSize: Int, path: String) throws -> Data {
+    guard expectedSize > 0 else { return Data() }
+
+    var output = Data(count: expectedSize)
```
Comment (UltralyticsAssistant) [inline Sources/YOLO/MiniZip.swift:219]: ⚠️ **HIGH**: `expectedSize` is taken directly from the archive metadata, and `Data(count:)` allocates that many bytes up front. A crafted ZIP can advertise a huge uncompressed size and force the app into an OOM before any validation happens. Since `download(from:)` accepts arbitrary URLs, this should be bounded or decompressed incrementally instead of trusting the header.

File: Sources/YOLO/MiniZip.swift [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+    let count = data.count
+    guard count >= 22 else { throw MiniZipError.notAZipFile }
+
+    // Locate the end-of-central-directory record by scanning backwards. The trailing comment may be up to
+    // 0xFFFF bytes, so search at most that far plus the 22-byte record itself.
+    let searchLimit = min(count, 22 + 0xFFFF)
+    var eocd = -1
+    var i = count - 22
+    while i >= count - searchLimit {
+      if u32(data, i) == endOfCentralDirectorySignature {
```
Comment (UltralyticsAssistant) [inline Sources/YOLO/MiniZip.swift]: 💡 **MEDIUM**: The backward EOCD scan accepts the first matching signature without validating that the candidate's comment length lands exactly at the end of the file. A valid ZIP whose trailing comment happens to contain `0x06054b50` can therefore be misidentified as corrupt. Please verify `candidate + 22 + commentLength == count` before accepting a match.

File: Sources/YOLO/MiniZip.swift [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+    switch entry.compressionMethod {
+    case 0:  // STORED — data is already uncompressed.
+      output = raw
+    case 8:  // DEFLATE — ZIP stores a raw DEFLATE stream, which COMPRESSION_ZLIB decodes.
+      output = try inflate(raw, expectedSize: entry.uncompressedSize, path: entry.path)
+    default:
+      throw MiniZipError.unsupportedFeature("compression method \(entry.compressionMethod)")
+    }
+
+    if entry.crc32 != 0, crc32(output) != entry.crc32 {
```
Comment (UltralyticsAssistant) [inline Sources/YOLO/MiniZip.swift]: 💡 **MEDIUM**: This skips CRC verification whenever the stored CRC is `0`, but `0` is a valid CRC-32 value (notably for empty files, and for some non-empty payloads). In those cases corrupted content will be accepted even though the archive provided a checksum.

**Suggested change:**
```suggestion
    if crc32(output) != entry.crc32 {
```

File: Sources/YOLO/MiniZip.swift (hunk 1 above)
Comment (glenn-jocher) [reply Sources/YOLO/MiniZip.swift:219]: ✅ Fixed in 344fa9e. `inflate(_:expectedSize:path:)` now rejects an entry before allocating when its advertised uncompressed size exceeds DEFLATE's maximum ~1032:1 expansion ratio for the actual compressed bytes (new `MiniZipError.entryTooLarge`). Since `compressedSize` is already bounded by the on-disk archive size, a crafted header can no longer force a large up-front allocation. Added `testDecompressionBombRejected` with a size-patched bomb fixture.

File: Sources/YOLO/MiniZip.swift (hunk 2 above)
Comment (glenn-jocher) [reply Sources/YOLO/MiniZip.swift]: ✅ Fixed in 344fa9e. The backward EOCD scan now additionally requires `candidate + 22 + commentLength == count`, so a `0x06054b50` appearing inside a trailing archive comment is skipped and the real record is found. Added `testCommentContainingSignatureIsHandled` with a fixture whose comment embeds the EOCD signature.

File: Sources/YOLO/MiniZip.swift (hunk 3 above)
Comment (glenn-jocher) [reply Sources/YOLO/MiniZip.swift]: ✅ Fixed in 344fa9e. Dropped the `entry.crc32 != 0` guard — the central directory always carries the real CRC-32 (including the legitimate 0 for empty files), so verification is now unconditional. Added `testEmptyFileEntryExtracts` covering a zero-byte (CRC 0) entry.

**Comment by UltralyticsAssistant** [comment]
Merged! 🎉 Huge thanks to @glenn-jocher for pushing the iOS app and YOLO package to a true zero-dependency design.

As Antoine de Saint-Exupéry said, “Perfection is achieved, not when there is nothing more to add, but when there is nothing left to take away.” This PR captures that perfectly by removing `ZIPFoundation`, introducing a secure native Swift extractor, and making the project simpler, safer, and easier to ship.

Really appreciate the thoughtful work here—from the built-in `MiniZip` implementation and security protections to the solid test coverage and documentation updates. A great step forward for maintainability and reliability in `yolo-ios-app`.
